server/opts.go

```diff
@@ -434,6 +434,21 @@ type WebsocketOpts struct {
 	// "jwt" specified in the CONNECT options is missing or empty.
 	JWTCookie string
 
+	// Name of the cookie, which if present in WebSocket upgrade headers,
+	// will be treated as Username during CONNECT phase as long as
+	// "user" specified in the CONNECT options is missing or empty.
+	UsernameCookie string
+
+	// Name of the cookie, which if present in WebSocket upgrade headers,
+	// will be treated as Password during CONNECT phase as long as
+	// "pass" specified in the CONNECT options is missing or empty.
+	PasswordCookie string
+
+	// Name of the cookie, which if present in WebSocket upgrade headers,
+	// will be treated as Token during CONNECT phase as long as
+	// "auth_token" specified in the CONNECT options is missing or empty.
+	TokenCookie string
```

Comment:
Should we encourage this? I feel it should not be used and in my mind is deprecated. WDYT?

Reply:
This is the perfect location for users to provide a non-nats JWT for callout to process.

Reply:
(this is why I thought this was a good thing to do)

Reply:
ok that make sense. So maybe we call that out in the comment, that this should not be a plain server wide token but  something external that folks want passed to the auth callout service etc.

Reply:
well, it can be anything they want, including a server-wide token even if that is discouraged.

Reply:
(added a good comment line)